docs/src/pages/_document.page.tsx

```diff
@@ -1,28 +1,39 @@
+import crypto from 'crypto';
 import Document, { Html, Head, Main, NextScript } from 'next/document';
 
 const favicon =
   process.env.NODE_ENV === 'development'
     ? '/svg/favicon-dev.svg'
     : '/svg/favicon.svg';
 
+const cspHashOf = (text) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(text);
+  return `'sha256-${hash.digest('base64')}'`;
+};
 class MyDocument extends Document {
   render() {
+    // See: https://github.com/vercel/next.js/blob/master/examples/with-strict-csp/pages/_document.js
+    let csp = `default-src 'self';
```

Comment:
can we put this logic into a separate function? I feel it adds unnecessary noise to the render function

Reply:
Sure, will update